pkg/actuators/machine/reconciler.go

```diff
@@ -342,6 +351,67 @@ func (r *Reconciler) updateLoadBalancers(instance *ec2.Instance) error {
 	return nil
 }
 
+// deletePlacementGroup deletes the placement group for the machine
+func (r *Reconciler) deletePlacementGroup() error {
+	placementGroupName := r.providerSpec.Placement.GroupName
+	if placementGroupName == "" {
+		// Nothing to do if the group name is empty
+		return nil
+	}
+
+	// Check that the placement group contains no instances
+	result, err := r.awsClient.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("placement-group-name"),
+				Values: []*string{aws.String(placementGroupName)},
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("could not get the number of instances in the placement group %s: %v", placementGroupName, err)
+	}
+
+	var instanceCount int
+	for _, reservation := range result.Reservations {
+		for range reservation.Instances {
+			instanceCount++
+		}
+	}
+	if instanceCount > 0 {
+		klog.Info("placement group for name %q still constains %d instances and cannot be removed", placementGroupName, instanceCount)
+		return nil
+	}
+
+	// Check that the placement group exists
+	placementGroups, err := r.awsClient.DescribePlacementGroups(&ec2.DescribePlacementGroupsInput{
+		GroupNames: []*string{aws.String(placementGroupName)},
+	})
+	if err != nil && !isAWS4xxError(err) {
```

Comment:
Just as a note, I might change how this error check works in my PR, some of the work on removing PGs in the installer made me think of a way to possibly make it better, I'll need to investigate this though